src/dlgTriggerEditor.cpp

```diff
@@ -3946,10 +3938,10 @@ void dlgTriggerEditor::saveVar()
     // we add a new variable
     // we switch away from a variable (so we are saving the old variable)
 
-    if (!mCurrentVar)
+    if (!mpCurrentVarItem)
         return;
-    QTreeWidgetItem * pItem = (QTreeWidgetItem*)mCurrentVar;
-    if (!pItem || !pItem->parent() )
+    QTreeWidgetItem * pItem = mpCurrentVarItem;
+    if (!pItem->parent())
```

Comment:
Oh, good catch!; it might be OK, but lets assume things can go wrong or the user can do things that we as developers don't consider _reasonable_!